--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/project/static/dashboard/project/workflow/launch-instance/launch-instance-model.service.spec.js~	2017-06-21 00:10:39.000000000 -0600
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/project/static/dashboard/project/workflow/launch-instance/launch-instance-model.service.spec.js	2017-11-28 14:08:31.905062124 -0700
@@ -863,8 +863,8 @@
           model.newInstanceSpec.security_groups = [ { id: 'adminId', name: 'admin' },
                                                     { id: 'demoId', name: 'demo' } ];
           model.newInstanceSpec.scheduler_hints = {};
-          model.newInstanceSpec.vol_create = true;
-          model.newInstanceSpec.vol_delete_on_instance_delete = true;
+          model.newInstanceSpec.vol_create = false;
+          model.newInstanceSpec.vol_delete_on_instance_delete = false;
           model.newInstanceSpec.vol_device_name = "volTestName";
           model.newInstanceSpec.vol_size = 10;
           model.newInstanceSpec.server_groups = [];
--- /usr/share/openstack-dashboard/openstack_dashboard/dashboards/project/static/dashboard/project/workflow/launch-instance/source/source.controller.js~	2017-06-21 00:10:39.000000000 -0600
+++ /usr/share/openstack-dashboard/openstack_dashboard/dashboards/project/static/dashboard/project/workflow/launch-instance/source/source.controller.js	2017-11-28 14:08:31.905062124 -0700
@@ -416,7 +416,7 @@
 
     function updateBootSourceSelection(selectedSource) {
       ctrl.currentBootSource = selectedSource;
-      if ((selectedSource === bootSourceTypes.IMAGE ||
+      if (false && (selectedSource === bootSourceTypes.IMAGE ||
            selectedSource === bootSourceTypes.INSTANCE_SNAPSHOT) && $scope.model.volumeBootable) {
         $scope.model.newInstanceSpec.vol_create =
           $scope.model.newInstanceSpec.create_volume_default;
