--- /usr/lib/python2.7/dist-packages/manila_ui/dashboards/project/shares/__init__.py~	2017-11-30 12:40:12.639482191 -0700
+++ /usr/lib/python2.7/dist-packages/manila_ui/dashboards/project/shares/__init__.py	2017-11-30 12:47:58.026021576 -0700
@@ -143,7 +143,7 @@
 
 
 @wrap(quotas.tenant_quota_usages)
-def tenant_quota_usages(f, request, tenant_id=None):
+def tenant_quota_usages(f, request, tenant_id=None, targets=None):
 
     usages = f(request, tenant_id, targets)
 
